src/containers/Content/index.js

```diff
@@ -0,0 +1,177 @@
+            orders: [],
+            ingredients: {},
+            orderedList: [],
+            total: "1.00",
+            isModalActive: false,
+            isPurchaseActive: false,
+            isLoading: true
+        }
+        
+        this.reload.bind(this);
+        this.addIngredientBuilder.bind(this);
+        this.addToBill.bind(this);
+        this.changeActiveModal.bind(this);
+        this.hideShowPurchase.bind(this)
+    }
+
+    componentDidMount = () => {
+        fetch("https://beetroot-burger-app.herokuapp.com/ingredients").then(res => res.json()).then(order => {
+            this.setState((oldState) => {
+                const orderFromBack = order[0].ingredients.map(e => {return {ingredient: e.name, price: e.price}});
+                let ingredientsFromBack = {};
+                orderFromBack.forEach((item) => {
+                    ingredientsFromBack[item.ingredient] = 0;
+                });
+                return {
+                    ...oldState,
+                    orders: orderFromBack,
+                    ingredients: ingredientsFromBack,
+                }
+            })
+        }).catch((err) => console.error(err)).finally(() => {
+            this.setState((oldState) => {
+                return {
+                    ...oldState,
+                    isLoading: false
+                }
+            })
+        })
+    }
+
+    addToState = (name, price) => {
+        this.setState((oldState) => {
+            return {
+                ...oldState,
+                total: (+oldState.total + +price).toFixed(2), 
+                ingredients: {
+                    ...oldState.ingredients,
+                    [name]: +oldState.ingredients[name] + 1
+                },
+                orderedList: [...oldState.orderedList, name]
+            }
+        })
+    }
+
+    removeFromState = (name, price) => {
+        let index = this.state.orderedList.lastIndexOf(name);
+        this.state.orderedList.splice(index, 1)
+        this.setState((oldState) => {
+            return {
+                ...oldState,
+                total: (+oldState.total - +price).toFixed(2), 
+                ingredients: {
+                    ...oldState.ingredients,
+                    [name]: +oldState.ingredients[name] - 1,
+                },
+                orderedList: oldState.orderedList
+            }
+        })
+    }
+
+    changeActiveModal = () => {
+        this.setState((oldState) => {
+            return {
+                ...oldState,
+                isModalActive: !oldState.isModalActive
+            }
+        });
+    }
+
+    addToBill = () => {
+        let li = "";
+        if (this.state.total > 0) {
+            Object.entries(this.state.ingredients).forEach(([key, value]) => {
+                let price = this.state.orders.filter((item) => item.ingredient === key)[0].price.toFixed(2);
+                if (value > 0) {
+                    li = [li, (<li key={key}>{key}: {value} x {price} $</li>)]
```

Comment:
Complicated logic and hardly maintainable code. Make it simpler. Avoid using loops inside loops